index.bs

```diff
@@ -1194,6 +1194,13 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
                         :   {{AuthenticatorAttestationResponse/attestationObject}}
                         ::  |attestationObject|
 
+                        :   {{AuthenticatorAttestationResponse/transports}}
+                        ::  A sequence of one or more {{AuthenticatorTransport}} values. The first value is set to the transport used to obtain this registration. Further values are appended to indicate other, distinct, transports that the |authenticator| is believed to support.
```

Comment:
> The first value is set to the transport used to obtain this registration

This means that there is a value in providing what transport is used back to the RP. By that logic, does is makes sense that RP can choose which transport it wants to use in the first place as an input parameter for both MakeCredential and GetAssertion??

Reply:
I don't think I care too much about the *specific* transport that was used during the operation, more about what's actually supported by the token, but if having both doesn't cause a privacy issue, I think it's good to have it.

>By that logic, does is makes sense that RP can choose which transport it wants to use in the first place as an input parameter for both MakeCredential and GetAssertion??
They can already choose during GetAssertion, but it's really more so the platform knows whether it can potentially satisfy the request (by doing an intersection between what it supports and what authenticators the user have registered).
During MakeCredential I'm not quite so sure it matters, apart from specifying the use case (which is why we have "attachment"). Is there a specific example which you think won't work using just "attachment" during create?

Reply:
> They can already choose during GetAssertion, but it's really more so the platform knows whether it can potentially satisfy the request (by doing an intersection between what it supports and what authenticators the user have registered).

No. You can't for username less flows as you don't know who the user is yet. That's why I always advocated for attachment in GetAssertion (which I will propose for next version as we ran out of time for this version). 

Adding a sequence of transport instead of attachment filter as a request option for both MakeCredential/GetAssertion makes it even more flexible. @christiaanbrand, @agl What do you think?

Reply:
> No. You can't for username less flows as you don't know who the user is yet. That's why I always advocated for attachment in GetAssertion.

I fear that I'm being slow and don't understand what you have in mind here. Would you mind expanding?

What I believe Christiaan has in mind here is the case where an RP requests an assertion and provides a credential that was created with a platform authenticator on another machine. If the RP can indicate that it's an &ldquo;internal&rdquo; transport then the browser can quickly establish that it doesn't have a matching private key and tell the user that. Otherwise it might prompt the user to insert their USB key or something, thus confusing them.

That obviously doesn't apply to usernameless logins however, so I imagine that you have some different scenario in mind that you want to address&mdash;I'm just not sure what it is.

Reply:
(This thread was continued at the top-level, below.)